internal/mode/static/handler.go

```diff
@@ -214,12 +225,36 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, logger logr.Log
 		}
 	}
 
+	h.latestReloadResult = nginxReloadRes
+
 	gwAddresses, err := getGatewayAddresses(ctx, h.cfg.k8sClient, nil, h.cfg.gatewayPodConfig)
 	if err != nil {
 		logger.Error(err, "Setting GatewayStatusAddress to Pod IP Address")
 	}
 
-	h.cfg.statusUpdater.Update(ctx, buildGatewayAPIStatuses(graph, gwAddresses, nginxReloadRes))
+	transitionTime := metav1.Now()
```

Comment:
This function is getting long. Can we extract a function here for all the status stuff?